public/app/features/explore/ContentOutline/ContentOutlineItemButton.tsx

```diff
@@ -1,87 +1,149 @@
 import { cx, css } from '@emotion/css';
-import React, { ButtonHTMLAttributes } from 'react';
+import React, { ButtonHTMLAttributes, useEffect, useRef, useState } from 'react';
 
 import { IconName, isIconName, GrafanaTheme2 } from '@grafana/data';
-import { Icon, useStyles2, Tooltip } from '@grafana/ui';
+import { Icon, useStyles2, Tooltip, IconSize } from '@grafana/ui';
+import { TooltipPlacement } from '@grafana/ui/src/components/Tooltip';
 
 type CommonProps = {
+  contentOutlineExpanded?: boolean;
   title?: string;
-  icon: string;
+  icon?: IconName | React.ReactNode;
   tooltip?: string;
+  tooltipPlacement?: TooltipPlacement;
   className?: string;
+  indentStyle?: string;
+  collapsible?: boolean;
+  collapsed?: boolean;
+  toggleCollapsed?: () => void;
   isActive?: boolean;
 };
 
 export type ContentOutlineItemButtonProps = CommonProps & ButtonHTMLAttributes<HTMLButtonElement>;
 
 export function ContentOutlineItemButton({
+  contentOutlineExpanded,
   title,
   icon,
   tooltip,
+  tooltipPlacement = 'bottom',
   className,
+  indentStyle,
+  collapsible,
+  collapsed,
+  toggleCollapsed,
   isActive,
   ...rest
 }: ContentOutlineItemButtonProps) {
   const styles = useStyles2(getStyles);
 
   const buttonStyles = cx(styles.button, className);
 
+  const textRef = useRef<HTMLElement>(null);
+  const [isOverflowing, setIsOverflowing] = useState(false);
+
+  useEffect(() => {
+    if (textRef.current) {
+      setIsOverflowing(textRef.current?.scrollWidth > textRef.current?.clientWidth);
+    }
+  }, []);
+
   const body = (
-    <button
-      className={cx(buttonStyles, {
-        [styles.active]: isActive,
-      })}
-      aria-label={tooltip}
-      {...rest}
-    >
-      {renderIcon(icon)}
-      {title}
-    </button>
+    <div className={cx(styles.buttonContainer, indentStyle)}>
+      {collapsible && (
+        <button
+          className={styles.collapseButton}
+          onClick={toggleCollapsed}
+          aria-label="content-outline-item-chevron-collapse"
```

Comment:
This should probably be more descriptive to users. probably should also be paired with `aria-expanded` and `aria-controls`. there are some good examples in https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded

Reply:
I've updated the description and added `aria-expanded`. Adding `aria-controls` would involve wrapping children outline items in ContentOutline in a div that would act as a section wrapper, and only have an id prop. Do you think this is necessary?

Reply:
yep, accessibility shouldn't be optional :)